ui/core/src/model/datasource.ts

```diff
@@ -11,22 +11,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Definition, JsonObject } from './definitions';
-import { Metadata } from './resource';
+import { Metadata, ProjectMetadata } from './resource';
 
-export interface DataSourceResource {
-  kind: 'DataSource';
-  metadata: Metadata;
-  spec: {
-    data_source: AnyDataSourceDefinition;
-  };
+export interface DatasourceSpecDefinition<Kind extends string> {
+  kind: Kind;
+  default: boolean;
 }
 
-export interface DataSourceDefinition<Kind extends string, Options extends JsonObject>
-  extends Definition<Kind, Options> {
-  display: {
-    hide?: boolean;
-  };
+export type AnyDatasourceSpecDefinition = DatasourceSpecDefinition<string>;
+
+export interface DatasourceModel {
+  kind: 'Datasource';
+  metadata: ProjectMetadata;
+  spec: AnyDatasourceSpecDefinition;
 }
 
-export type AnyDataSourceDefinition = DataSourceDefinition<string, JsonObject>;
+export interface GlobalDatasourceModel {
+  kind: 'GlobalDatasource';
+  metadata: Metadata;
+  spec: AnyDatasourceSpecDefinition;
+}
```

Comment:
Not sure I understand the difference between "global" and not global?

Reply:
A `GlobalDatasource` is a datasource that can be used by any dashboards the different `project`.
A not global `Datasource` is a datasource scoped to a `project`. So only the dashboards that are part of the same project than the not global `Datasource` can use it.

Reply:
For the moment the not global `Datasource`s are not supported. Since they are scoped to a project selected, that means the list will change you are changing the project.
I think it will be nice if we can change the way to get the datasources. Like it's simply fetch the datasources when you go on a dashboard page. 
If the code is fetching the `GlobalDatasources` then it put in a cache. Like that it won't be fetched everytime you are changing the dashboard, which was the objective of the datasource registering. 

Would you be ok with this approach ?